prowler/lib/check/check.py

```diff
@@ -356,6 +358,13 @@ def execute_checks(
         audit_progress=0,
     )
 
+    # Check ulimit for the maximum system open files
+    soft, _ = getrlimit(RLIMIT_NOFILE)
+    if soft < 4096:
+        logger.warning(
+            f"Prowler may have errors because of your user session maximum open files ({soft}). To solve this issue, increase the shell session limit by running this command `ulimit -n 4096`. More info in https://docs.prowler.cloud/en/latest/troubleshooting/"
+        )
+
```

Comment:
Why don't you include this at the beginning of the Prowler execution?

Reply:
Because it does not make sense to take it into account when there is no checks to be executed.
For example, when listing variables.

Reply:
Totally agree, thanks!